components/library/openssl/openssl-3.1/patches/043-x86_asm_illegal_subtraction.patch

```diff
@@ -0,0 +1,32 @@
+#
+# This was developed in house to fix x86 compilation of hand crafted
+# assembler, namely optimized AES cipher implementations.
+# Otherwise, with Solaris assembler, the build will fail with errors
+# similar to this one:
+#
+#   "crypto/aes/aes-586.s", line 1002 : Illegal subtraction in ... "OPENSSL_ia32cap_P - .L004pic_point"
+#
+# The problem is the reference to global variable (OPENSSL_ia32cap_P)
+# which Oracle Developer Studio assembler cannot deal with. The assembler label references
+# are fine.
+#
+# There is an upstream bug that matches this:
+#   https://github.com/openssl/openssl/issues/1715
+#
+# However it was closed with advice to use newer/better (i.e. GNU) assembler.
+#
+# This is tracked as enhancement for Oracle Developer Studio:
+#
+#     30005149 treat undefined subtraction operand in leal instruction as relocation entry
```

Comment:
I think we do not need this patch since it looks Studio specific.

Reply:
Correct this might be an issue on sparc but I am not looking into too many things here.